src/connectedComponents/ToolbarRow.js

```diff
@@ -192,6 +192,32 @@ function _getButtonComponents(toolbarButtons, activeButtons) {
   });
 }
 
+/**
+ * Checks whether the passed button is currently active
+ *
+ * @param {*} button
+ */
+function _isButtonAlreadyActive(button) {
+  const { activeButtons } = this.state;
+  return activeButtons.includes(button.id);
+}
+
+/**
+ * Sets the tool associated to the active button as passive
+ */
+function _setActiveToolAsPassive() {
+  const { activeButtons, toolbarButtons } = this.state;
+
+  const activeButtonId = activeButtons[0];
```

Comment:
I think it would be better to pass the clicked button in to this instead of assuming it lives at `activeButtons[0]`.

Reply:
The clicked button is not the one that is active though. This happens before the clicked button is set into state.

Reply:
Still seems a bit questionable to disable the button at `activeButtons[0]`

Reply:
Sure, I'll check whether it exists. I wanted to make sure we were in the same page though.

Reply:
Updated! I'd appreciate reviewing this changes if possible

Reply:
This looks alright. But as I was thinking about what's going on here, it might make sense to disable all the tools in the list rather than assuming the 0th one is the correct and only tool that needs disabling.

Other than that, I think this will work great for our purposes.

For the broader ohif
--
This seems like some details are bleeding through in to ToolbarRow, it probably shouldn't know what's necessary to deactivate a button or command - even as detailed as which `commandModule.js` command needs to be run. The activate/deactivate logic should probably be part of the configuration passed in through things like `toolbarModule.js`. Perhaps widening it to have an `activateCommandName` (should probably be a convention that implies the possibility of single use) and a `deactivateCommandName` where deactivate command name could either be unpopulated or a noop. ToolbarRow 

Then `_handleToolbarButtonClick` could be something like:
```
activeButtons.forEach(activeButton => {
    const deactivateOptions = Object.assign({ evt }, activeButton.deactivateCommandOptions);
    commandsManager.runCommand(activeButton.deactivateCommandName, deactivateOptions);
});

commandsManager.runCommand(button.activateCommandName, options);
```

This way, we're supporting any number of tool or action types that would be activated or deactivated, not just cornerstoneTools, and none of the details of that seep in to the ToolbarRow.

Reply:
Yeah that makes sense. I'll follow up on this with another PR such that we can start playing with the tool for now.